### ISSUE TITLE
Implement average marginal effects for brms models

### ISSUE THREAD

**Issue by jackobailey** [paul-buerkner/brms#767]
For models other than linear regression, interpreting model coefficients is hard. This is often because they refer to abstract scales (logits, probits) rather than that of the outcome we're interested in (probabilities, counts). One way of overcoming this is to compute *average marginal effects*. Reporting these is near standard in some fields (economics, political science).

@leeper has implemented these for many model types in his [margins](https://github.com/leeper/margins) package. He has also written [a good accompanying paper](https://cran.r-project.org/web/packages/margins/vignettes/TechnicalDetails.pdf) that explains marginal effects and how to calculate them.

Margins does not work with brms. This is unfortunate, as brms offers unparalleled opportunities when it comes to estimating non-linear models. Thus, I propose that you implement computing average marginal effects in brms.

I have had to compute marginal effects for brms models for a project that I am working on. [Here's my attempt at a first pass](https://gist.github.com/jackobailey/0982c89326c36b12d6fa6d6f182189be) based on @leeper's code. As you can see, the code is relatively straightforward and, as it relies on the `fitted` function, suitably generic as to be extendable to most model types.

**Comment by MilaniC** [comment]
Is this not the same as the open issue #552? "How to marginalize rather than condition on variables to make the output of brms marginal_effects literal AME, MER, and MEM feature (opened on 15 Nov 2018 by realkrantz).

**Comment by paul-buerkner** [comment]
This is true. I forget that myself, I am afraid... I just cited @jackobailey's text in the other issue as I think it is very helpful and will close this issue to avoid duplication.
